src/components/Communication.js

```diff
@@ -31,23 +31,36 @@ class Communication {
   /*
    * Called when the user has tried to login to deepstream.
    * Will be used later for exception handling
-   * eslint-disable no-unused-vars
-   * eslint-disable class-methods-use-this
   */
+  /* eslint-disable */
   onLoggedIn(success, data) {}
+  /* eslint-enable */
 
   /*
    * Request the instances that are currently running.
    */
-  requestInstances(callback) {
-    this.client.rpc.make('services/getInstances', {}, callback);
+  requestInstances(onInstancesReceived, onPlayerAdded, onInstanceCreated) {
+    this.client.rpc.make('services/getInstances', {}, onInstancesReceived);
+    this.client.event.subscribe('services/playerAdded', data => {
+      onPlayerAdded(data.playerName, data.instanceName);
+    });
+    this.client.event.subscribe('services/instanceCreated', data => {
+      onInstanceCreated(data.name);
+    });
+  }
+
+  stopRequestInstances() {
+    this.client.event.unsubscribe('services/playerAdded');
+    this.client.event.unsubscribe('services/instanceCreated');
   }
 
   /*
    * Join a certain instance with the given name.
    */
-  joinInstance(instanceName, callback) {
+  joinInstance(instanceName, name, callback) {
     this.instance = instanceName;
+    this.name = name;
+    console.log(callback);
```

Comment:
Ska denna ligga kvar här?

Reply:
```javascript
this.name = name;
```
This sets the username to the name that the player wrote into the UserInput screen.

```javascript
console.log(callback);
```
Yes, this should be removed.